lib/topological_inventory/openshift/operations/core/service_catalog_client.rb

```diff
@@ -83,14 +87,33 @@ def service_instance_provisioning?(service_instance)
           end
 
           def connection
-            Thread.current[:kubernetes_connection] ||= connection_manager.connect(
-              "servicecatalog", :host => default_endpoint.host, :token => authentication.password, :verify_ssl => verify_ssl_mode
-            )
+            Thread.current[:kubernetes_connection] ||= begin
+              raise "Unable to find a default endpoint for source [#{source_id}]" if default_endpoint.nil?
+              raise "Unable to find an authentication for source [#{source_id}]"  if authentication.nil?
+
+              connection_manager.connect("servicecatalog", :host => default_endpoint.host, :token => authentication.password, :verify_ssl => verify_ssl_mode)
+            end
           end
 
           def verify_ssl_mode
             default_endpoint.verify_ssl ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE
           end
+
+          def fetch_default_endpoint
+            endpoints = sources_api_client.list_source_endpoints(source_id)&.data || []
+            endpoints.find(&:default)
+          end
+
+          def fetch_authentication
+            endpoint = default_endpoint
+            return if endpoint.nil?
+
+            endpoint_authentications = sources_api_client.list_endpoint_authentications(endpoint.id.to_s).data || []
+            return if endpoint_authentications.empty?
+
+            auth_id = endpoint_authentications.first.id.to_s
```

Comment:
This `to_s` wasn't there before. It doesn't look needed: https://github.com/ManageIQ/topological_inventory-openshift/blob/master/lib/topological_inventory/openshift/operations/core/authentication_retriever.rb#L9

Reply:
Done